packages/url_launcher/url_launcher_ios/lib/url_launcher_ios.dart

```diff
@@ -45,11 +47,39 @@ class UrlLauncherIOS extends UrlLauncherPlatform {
     required bool universalLinksOnly,
     required Map<String, String> headers,
     String? webOnlyWindowName,
-  }) {
+  }) async {
+    final LaunchResultDetails result =
+        await _launchUrl(useSafariVC, url, universalLinksOnly);
+    return _mapLaunchResults(results: result);
+  }
+
+  Future<LaunchResultDetails> _launchUrl(
+      bool useSafariVC, String url, bool universalLinksOnly) {
     if (useSafariVC) {
       return _hostApi.openUrlInSafariViewController(url);
     } else {
       return _hostApi.launchUrl(url, universalLinksOnly);
     }
   }
+
+  bool _mapLaunchResults({required final LaunchResultDetails results}) {
+    switch (results.result) {
+      case LaunchResult.success:
+        return true;
+      case LaunchResult.failure:
+        return false;
+      case LaunchResult.invalidUrl:
```

Comment:
This would need explanatory comments with a link to https://github.com/flutter/flutter/issues/127665